code/game/machinery/defibrillator_mount.dm

```diff
@@ -23,20 +23,20 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/defibrillator_mount, 28)
 /obj/machinery/defibrillator_mount/loaded/Initialize(mapload) //loaded subtype for mapping use
 	. = ..()
 	defib = new/obj/item/defibrillator/loaded(src)
+	RegisterSignal(defib, COMSIG_QDELETING, PROC_REF(on_defib_del))
 
 MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/defibrillator_mount, 28)
 
 /obj/machinery/defibrillator_mount/Destroy()
 	if(defib)
 		QDEL_NULL(defib)
-	. = ..()
-
-/obj/machinery/defibrillator_mount/handle_atom_del(atom/A)
-	if(A == defib)
-		defib = null
-		end_processing()
 	return ..()
 
+/obj/machinery/defibrillator_mount/proc/on_defib_del(atom/source)
+	SIGNAL_HANDLER
+	defib = null
+	end_processing()
```

Comment:
Why can't this be exited, I feel like we can serve dual purpose of handing defib removal and defib deletion with it

Reply:
I had confused the defib for its paddles, which are what gets put in the user's hands, I'll fix that.